libcxx/include/__memory/is_sufficiently_aligned.h

```diff
@@ -0,0 +1,34 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _LIBCPP___MEMORY_IS_SUFFICIENTLY_ALIGNED_H
+#define _LIBCPP___MEMORY_IS_SUFFICIENTLY_ALIGNED_H
+
+#include <__config>
+#include <__cstddef/size_t.h>
+#include <cstdint>
+
+#if !defined(_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER)
+#  pragma GCC system_header
+#endif
+
+_LIBCPP_BEGIN_NAMESPACE_STD
+
+#if _LIBCPP_STD_VER >= 26
+
+template <size_t _Alignment, class _Tp>
```

Comment:
I think we can just add this to `assume_aligned.h`. These two seem to be quite closely related.

Reply:
My slight preference is to keep it in a separate header.

Reply:
I don't think they're just related as "alignment stuff". `is_sufficiently_aligned` is the precondition for `assume_aligned`. If `is_sufficiently_aligned` wasn't (IMO unnecessarily) non-`constexpr` I'd be strongly in favor of checking the precondition that way.

Reply:
> If `is_sufficiently_aligned` wasn't (IMO unnecessarily) non-`constexpr` I'd be strongly in favor of checking the precondition that way.

I don't know that it is implementable.  How would you check alignment in a constant expression?

Reply:
Ack, I also don't see how this can be implemented inside a constant expression.

Reply:
https://godbolt.org/z/7xGqasbGz is one implementation.

Reply:
Using `__builtin_constant_p` like this is a bad idea; it won't produce  the effects required by the standard.
See https://godbolt.org/z/jerx3KjW4